BootloaderCorePkg/Stage2/Stage2.c

```diff
@@ -391,6 +391,10 @@ SecStartup (
   FspResetHandler (Status);
   ASSERT_EFI_ERROR (Status);
 
+  if (FixedPcdGetBool (PcdSmbiosEnabled)) {
+    InitSmbiosStringPtr ();
+  }
+

```

Comment:
Had better initialize SMBIOS string in the SmbiosInit() of SmBiosLib.

Reply:
Hi Guo,
 InitSmbiosStringPtr () is placed in Stage2 before PostSiliconInit state because so that customer can override the PcdSmbiosStringsPtr with custom SMBIOS Strings in the platform specific Boardinit in PostSiliconInit state and also Prior to this we need to do memory allocation of PcdSmbiosStringsPtr and assign default SMBIOS strings to it.
This will also dont affect current implementation in Coffeelake and Apollolake.

Reply:
Yes, this needs to be done before SMBIOS init to allow platform customization.  SmbiosInit() will create the final SMBIOS table.

Reply:
ok. Let's use this one firstly.